File: python/765.py
from typing import List

class Solution:
    def minSwapsCouples(self, row: List[int]) -> int:
        n = len(row)
        pos = [0] * n
        for index, p in enumerate(row):
            pos[p] = index
        ans = 0
        for i in range(0, n , 2):
            if row[i] ^ row[i + 1] != 1:
                ans += 1
                row[pos[row[i] ^ 1]] = row[i+1]
                pos[row[i+1]] = pos[row[i] ^ 1]
        return ans

File: go/765.go
package main

import "fmt"

func minSwapsCouples(row []int) (ans int) {
	n := len(row)
	pos := make([]int, n)
	for index, p := range row {
		pos[p] = index
	}
	for i := 0; i < n; i += 2 {
		if row[i]^row[i+1] != 1 {
			ans++
			row[pos[row[i]^1]] = row[i+1]
			pos[row[i+1]] = pos[row[i]^1]
		}
		fmt.Printf("row: %v\n", row)
	}
	return
}

// func main() {
// 	row := []int{0, 2, 4, 6, 7, 1, 3, 5}
// 	minSwapsCouples(row)
// }
